hw1/02-palindrome.js

```diff
@@ -1,3 +1,40 @@
-const elem = document.querySelector('input');
+const input = document.querySelector("input");
+const result = document.querySelector("#result");
 
-elem.addEventListener('input', handleInput);
+input.addEventListener("input", () => {
+  const value = input.value;
+
+  if (value < 0 || isNaN(value)) {
+    result.classList.remove("text-success");
+    result.classList.add("text-danger");
+    result.textContent = "Invalid input";
+    return;
+  }
+
+  if (value === "") {
+    result.textContent = "";
+    return;
+  }
+
+  const isPalindrome = isNumberPalindrome(value);
+  result.textContent = isPalindrome
+    ? "Yes! This is a palindrome!"
+    : "No, Try again.";
+});
+
+function isNumberPalindrome(num) {
+  const str = String(num);
```

Comment:
Curious but why don't we just pass the string version of the input instead of passing a number and convert it to string again ❓